src/mpi/comm/comm_split_type.c

```diff
@@ -860,6 +1010,10 @@ int MPIR_Comm_split_type_network_topo(MPIR_Comm * comm_ptr, int key, const char
                && *(hintval + strlen("min_mem_size:")) != '\0') {
         long min_mem_size = atol(hintval + strlen("min_mem_size:"));
         mpi_errno = network_split_by_min_memsize(comm_ptr, key, min_mem_size, newcomm_ptr);
+    } else if (!strncmp(hintval, ("torus_size:"), strlen("torus_size:"))
+               && *(hintval + strlen("torus_size:")) != '\0') {
+        int min_size = atol(hintval + strlen("torus_size:"));
+        mpi_errno = network_split_by_min_torus_size(comm_ptr, key, min_size, newcomm_ptr);
```

Comment:
How is `torus_size` different from `subcomm_min_size`?

Reply:
Torus size is based on the network properties, and assumes uniform distribution of processes among all nodes of the network.  ```subcomm_min_size``` makes no assumptions about the number of MPI ranks/node in the network.

Reply:
Why would the user care about whether it's a torus or a fat tree or some other topology?  Wouldn't they want just a collection of processes that are "close to each other"?  In that sense, `subcomm_min_size` is all we need: that needs to be applied to fat-tree or torus or other topologies internally without exposing that information to the user.

Reply:
That makes sense, I'll get rid of this. We decided to go with switch-level hint for fat trees, which is also covered by subcomm_min_size, do you think I should get rid of that as well?

Reply:
No, I think the `switch-level` hint is more like "give me all the processes on the first level of the switch".  It doesn't specify how many processes the application needs.  An equivalent for the torus would be something like "give me all the processes that share the x-dimension of the torus" (which we can provide with the current topology information that we already query), or "give me all the processes that are on the same rack" (which might be harder to query with the current hwloc/netloc).

Reply:
Torus size is not exactly in the same vein, but it also does not specify how many processes the application needs, but covers as many nodes as the user requests, while maximizing the covered dimensions. But I get your point about how this might be very topology aware and obscure to the user. I will remove the torus size hint and add torus dimension hint instead.

Reply:
I have updated the patches for torus dimension instead of torus size.